staging/src/k8s.io/apimachinery/pkg/watch/mux.go

```diff
@@ -209,16 +210,30 @@ func (m *Broadcaster) closeAll() {
 	m.watchers = map[int64]*broadcasterWatcher{}
 }
 
-// Action distributes the given event among all watchers.
+// Action distributes the given event among all watchers. It panics if Shutdown()
+// has already been called.
 func (m *Broadcaster) Action(action EventType, obj runtime.Object) {
 	m.incoming <- Event{action, obj}
```

Comment:
Why don't we check m.stopped in in Action as well?

Reply:
I didn't want to change the behavior of the function. `ActionOrDrop` already assumes that events could be dropped under specific conditions. 
I can add the check as to `Action` as well.

Reply:
Shutdown() documents "You must not call Action or Watch* after calling Shutdown.".

If we want to make it safe to call `Action*` or `Watch*` after calling Shutdown(), we need to do more than check the stopped channel in a racy way in those methods.